web/packages/shared/components/TerminalSearch/TerminalSearch.tsx

```diff
@@ -0,0 +1,180 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useEffect, useState, useRef, useCallback } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { Flex, ButtonIcon, Box, P2, Input } from 'design';
+import { ArrowDown, ArrowUp, Cross } from 'design/Icon';
+import { SearchAddon } from '@xterm/addon-search';
+
+export interface TerminalWithSearch {
+  getSearchAddon({
+    onDidChangeResults,
+  }: {
+    onDidChangeResults: ({
+      resultIndex,
+      resultCount,
+    }: {
+      resultIndex: number;
+      resultCount: number;
+    }) => void;
+  }): SearchAddon;
+}
+
+export const TerminalSearch = ({
+  xTerm,
+  show,
+  onClose,
+  onOpen,
+  focusTerm,
+  isSearchKeyboardEvent,
+}: {
+  xTerm: TerminalWithSearch;
+  show: boolean;
+  focusTerm(): void;
+  onClose(): void;
+  onOpen(): void;
+  isSearchKeyboardEvent: (e: KeyboardEvent) => boolean;
+}) => {
+  const theme = useTheme();
+  const searchInputRef = useRef<HTMLInputElement>();
+  const [searchValue, setSearchValue] = useState('');
```

Comment:
i think i feel the same. i had state used for the next/previous buttons, but i like the idea of using the ref only here. I'll change it!

Reply:
I forgot, the value is there so that the value of the input stays if we do somethig like
1. open search and type something
2. press escape
3. open search again

the value should be what it was before you closed it the first time (you can test this functionality that it should match close to what the browser in chrome does)

Reply:
Oh I see it. Another small UX improvement would be to take the selected text as the input value (Chrome does this too). xterm allows to read it through `getSelection()`. But I'm not requiring implementing it.

Reply:
I actually do this in web already, i must have missed it in Connect. I'll add!